lib/solveEquation/stepThrough.js

```diff
@@ -68,7 +71,42 @@ function stepThrough(leftNode, rightNode, comparator, debug=false) {
     }
 
     try {
-      equationStatus = step(equation, symbolName);
+      // Find roots of a equation
+      // Conditions: left side is a mul node, right side is 0
+      // TODO: move polynomial to left if is on the right
+      // TODO: assuming a single symbol in equation
+      // TODO: display all solutions
+      // Returns: An equation in the form of symbol = root
+      if (equation.leftNode.op === '*'
+          && (Node.Type.isConstant(equation.rightNode) && parseFloat(equation.rightNode.value) === 0)) {
+        // Gets all factors that aren't a constant
+        const factors = equation.leftNode.args.filter(arg => !Node.Type.isConstant(arg));
+
+        const solutions = [];
+
+        let symbol;
+        // For each factor, solve the equation factor = 0 for the symbol
+        for (var f in factors) {
+          const leftNode = factors[f].content;
+          const rightNode = Node.Creator.constant(0);
+
+          const lastStep = stepThrough(leftNode, rightNode, '=').slice(-1)[0];
+
+          symbol = lastStep.newEquation.leftNode;
+          // Append to a list of roots
+          solutions.push(lastStep.newEquation.rightNode);
+        }
+
+        const roots = new Equation(symbol, solutions[0], '=');
+        steps.push(new EquationStatus(
+          ChangeTypes.FIND_ROOTS, equation, roots));
+
+        // Return because we are done at this point
+        return steps;
+      }
+      else {
+        equationStatus = step(equation, symbolName);
+      }
     }
     catch (e) {
       // This error happens for some math that we don't support
```

Comment:
if you decide to keep this hacky code, update this comment at least noting that it will catch a whole bunch of errors, not just "some math we don't support"

edit: oh, I just expanded the catch block and it bubbles up other errors - which is good - but it's gonna be throwing a *lot* of errors for things that don't match exactly what you want (for all the things I commented on above)